File: dotnet/src/Datastructure/Otterop/Datastructure/ListIterator.cs
namespace Otterop.Datastructure
{
    class ListIterator<T> : Otterop.Lang.OOPIterator<T>
    {
        private List<T> list;
        private int index;
        internal ListIterator(List<T> list)
        {
            this.list = list;
            this.index = 0;
        }

        public bool HasNext()
        {
            return this.index < this.list.Size();
        }

        public T Next()
        {
            T ret = this.list.Get(this.index);
            this.index++;
            return ret;
        }

    }

}

File: java/datastructure/src/main/java/otterop/datastructure/ListIterator.java
package otterop.datastructure;

import otterop.lang.OOPIterator;

class ListIterator<T> implements OOPIterator<T> {

    private List<T> list;
    private int index;

    ListIterator(List<T> list) {
        this.list = list;
        this.index = 0;
    }

    public boolean hasNext() {
        return this.index < this.list.size();
    }

    public T next() {
        T ret = this.list.get(this.index);
        this.index++;
        return ret;
    }
}
